tools/src/icon4pytools/py2fgen/wrappers/diffusion_wrapper.py

```diff
@@ -297,6 +304,13 @@ def grid_init(
     v2c: gtx.Field[gtx.Dims[dims.VertexDim, dims.V2CDim], gtx.int32],
     e2c2v: gtx.Field[gtx.Dims[dims.EdgeDim, dims.E2C2VDim], gtx.int32],
     c2v: gtx.Field[gtx.Dims[dims.CellDim, dims.C2VDim], gtx.int32],
+    c_owner_mask: gtx.Field[[dims.CellDim], bool],
+    e_owner_mask: gtx.Field[[dims.EdgeDim], bool],
+    v_owner_mask: gtx.Field[[dims.VertexDim], bool],
+    c_glb_index: gtx.Field[[SpecialADim], gtx.int32],
```

Comment:
can it be that these are the "real" number of (cells, vertices, edges)? I faintly remember something like that from the discussion with @abishekg7 . If this is (reliably) you could determine the `num_cells`, `num_edges`, `num_vertices` from the shape of the `_glb_index` arrays.

Reply:
I am unsure if these arrays will give us the `real` numer of cells, edges and vertices, but I can take a look. Since these changes based on each "portion" of the decomposed grid as far as I understand I'd feel more comfortable taking them out from the `icon_grid` object itself, isn't this good enough?